app/celery/tasks.py

```diff
@@ -294,11 +295,13 @@ def save_api_email_or_sms(self, encoded_notification):
     provider_task = (
         provider_tasks.deliver_email if notification["notification_type"] == EMAIL_TYPE else provider_tasks.deliver_sms
     )
+
     try:
         persist_notification(
             notification_id=notification["id"],
             template_id=notification["template_id"],
             template_version=notification["template_version"],
+            template_has_unsubscribe_link=notification["template_has_unsubscribe_link"],
```

Comment:
i just realised, we can reuse the SerialisedTemplate code from `save_email` above - and then we can just use `template.has_unsubscribe_link` here and we won't need to pass it through the encoded notification stuff! that'll make this a lot simpler and save us editing a whole bunch of tests 🙏

https://github.com/alphagov/notifications-api/blob/8353ed0fd15549407ad45b6c491bfcbb2b2e5f41/app/celery/tasks.py#L238-L242

```suggestion
            template_has_unsubscribe_link=template.has_unsubscribe_link,
```

Reply:
also this is the code that i'm proposing to remove so hopefully this whole function can go away soon and then we dont have to change this 🤞

Reply:
@leohemsted 
I had considered it previously, at the time it seemed more efficient to add the `template.has_unsubscribe_link` value to `encoded_notification` rather than pulling in the `SerialisedTemplate` just to extract the same value.
Since `save_api_email` is about to be removed, I think I will remove this code and all the changes made to support it and not deploy the work done in this PR until after `save_api_email` has been removed.